src/backend/input.rs

```diff
@@ -216,6 +231,9 @@ pub trait InputBackend: Sized {
     /// Get current `InputConfig`
     fn input_config(&mut self) -> &mut Self::InputConfig;
 
+    /// Called to inform the Input backend about a new focused Output for a `Seat`
+    fn set_output_metadata(&mut self, seat: &Seat, output: &Output);
```

Comment:
I am think about make much more then three types out of them.

Not being able to match the `PointerEvent` feels so wrong. Then every method would return an `Option`,  because e.g. `PointerAxisEvents` do not have a `motion` or `position` property.

So I would go with associated types for:
`KeyboardKeyEvent`
`PointerAxisEvent`
`PointerButtonEvent`
`PointerMotionEvent`
`PointerMotionAbsoluteEvent`
`TouchDownEvent`
`TouchUpEvent`
`TouchMotionEvent`
`TouchCancelEvent`
`TouchFrameEvent`